Packages/ClientRuntime/Sources/Serialization/SerializationUtils/TimestampSerdeUtils.swift

```diff
@@ -0,0 +1,236 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Custom timestamp serialization formats
+/// https://awslabs.github.io/smithy/1.0/spec/core/protocol-traits.html#timestampformat-trait
+public enum TimestampFormat: CaseIterable {
+    /// Also known as Unix time, the number of seconds that have elapsed since 00:00:00 Coordinated Universal Time (UTC), Thursday, 1 January 1970, with optional fractional precision (for example, 1515531081.1234).
+    case epochSeconds
+    
+    /// Date time as defined by the date-time production in RFC3339 section 5.6 with no UTC offset and optional fractional precision (for example, 1985-04-12T23:20:50.52Z).
+    case dateTime
+    
+    /// An HTTP date as defined by the IMF-fixdate production in RFC 7231#section-7.1.1.1 (for example, Tue, 29 Apr 2014 18:30:38 GMT) with optional fractional seconds (for example, Sun, 02 Jan 2000 20:34:56.000 GMT).
+    case httpDate
+}
+
+// MARK: - Encoding Helpers
+
+struct TimestampEncodable: Encodable {
```

Comment:
Will add tests and docs for this. Wanted to get this up quickly to verify it fixes issues running swift 5.5.3 (confirmed locally)

Reply:
tests and docs have been added